checkmk_extensions/agent/scheduler_runner.ps1

```diff
@@ -0,0 +1,63 @@
+# Enable StrictMode 3.0
+Set-StrictMode -Version 3.0
+
+class Config {
+    [bool]$rcc
+
+    Config() {
+        # Default constructor
+        # By default, we will not run in RCC
+        $this.rcc = $false
+    }
+
+    static [string] GetConfigContent($configFilePath, [ref]$errorCollection) {
+        try {
+            if (-Not (Test-Path -Path $configFilePath -PathType Leaf)) {
+                # Add the error message to the error collection
+                $errorCollection.Value.Add("Config file not found: $configFilePath")
+
+                # If we want to throw an ERRORRECORD object or a .NET exception, we can use the something like:
+                # $errorCollection.Value.Add((New-Object System.IO.FileNotFoundException))
+                return $null
+            }
+
+            $configFileContent = Get-Content -Raw -Path $configFilePath
+            return $configFileContent
+        }
+        catch {
+            # Add the error message to the error collection
+            $errorCollection.Value.Add($_.Exception.Message)
+            return $null
+        }
+    }
+
+    # Method to parse the config file (JSON) and set the value of rcc
+    static [Config] ParseConfigFile($configFilePath) {
+        $config = [Config]::new()
+        $errors = [System.Collections.Generic.List[string]]::new()
+
+        try {
+            $configFileContent = [Config]::GetConfigContent($configFilePath, ([ref]$errors))
+            if ($null -ne $configFileContent -and $configFileContent -ne '') {
+                $configData = ConvertFrom-Json $configFileContent
+                # Check if it contains the rcc propetry and if it's equal to YES
+                if ($configData.PSobject.Properties["rcc"] -and $configData.rcc -eq "YES") {
```

Comment:
Can't we just do
`
$config.rcc = $configData.PSobject.Properties["rcc"]
`
Or, to be safe (not sure if this is proper Powershell syntax)
`
$config.rcc = $configData.PSobject.Properties["rcc"] -as [bool]
`

Reply:
I was only able to make this work, if in the json file the value for `rcc` is either `true` or `false`.

In all other cases, for example if we use `Yes` and `No`, it will evaluate the existance of a string as `true`, so it will lead to false positives.

Since we can influence the bakery rule, we can make sure that the value in the json will be a `bool`.

Reply:
That's fine, let's use `-as [bool]` (similar to what pydantic does) and if that leads to a false positive, it's definitely not our fault